artiq/firmware/ksupport/rtio.rs

```diff
@@ -149,6 +156,38 @@ mod imp {
         }
     }
 
+    pub extern fn input_timestamped_data(timeout: i64, channel: i32) -> TimestampedData {
```

Comment:
Nitpicking but IIRC you can put ``#[repr(C)]`` on the function and return ``(u64, i32)``. This way it's a tuple in both Rust and Python; and not a structure in Rust and a tuple in Python.